string destination_node_name
string source_node_name
---